### PR TITLE
snap: add new `snap whoami --refresh-keys` to update ssh keys

### PR DESCRIPTION
**Pull request by mvo5**
[Stawman]

This commit adds support for manually refresh the ssh keys of
the current user from Ubuntu SSO.

This is a first step to help with:
https://bugs.launchpad.net/snapd/+bug/1646559
so that at least after adding a key it's trivial to refresh it.
Users could enable a systemd user time to do this periodically
too.

Doing the real automatic syncing requires a bit more thinking

Doing the real automatic syncing requires a bit more thinking
(it seems to me) because then we would need to think about how
to detect if keys are managed by non-snapd and how to avoid
sending too many SSO requests etc.

### CODE REVIEW

**Comment by pedronis** [comment]
fwiw we have many kind of keys, so maybe it should be --refresh-ssh-keys

**Comment by mvo5** [comment]
Thanks! I updated this to `--refresh-ssh-keys` happy to add proper tests if there is a chance this could get accepted.

File: cmd/snap/cmd_whoami.go [hunk 1]
```diff
@@ -34,12 +40,47 @@ The whoami command shows the email the user is logged in with.
...
+
+	// FIXME: set auth context
+	var storeCtx store.DeviceAndAuthContext
+	sto := storeNew(nil, storeCtx)
+	userInfo, err := sto.UserInfo(email)
+	if err != nil {
+		return fmt.Errorf("cannot refresh keys for %q: %s", email, err)
+	}
+	if len(userInfo.SSHKeys) == 0 {
+		return fmt.Errorf("cannot reresh keys for %q: no ssh keys found", email)
```
Comment (mardy) [inline cmd/snap/cmd_whoami.go:68]: typo: refresh

Though maybe it's better to remove all these "cannot refresh keys for %q" prefixes from the error message, and add them in the place where this function is called? Not sure if there's a convention in the golang community about how error messages should be formatted.

Comment (mvo5) [reply cmd/snap/cmd_whoami.go]: I tweak the code a bit more

**Pull request closed without merging**